### PR TITLE
feat(lib): Add validation against accidental backend reuse

### PR DESCRIPTION
**Pull request by ansgarm**
depends on #2591

- fix(lib): fix cause for circular dependency in tests when importing app
- feat(lib): Add validation for accidental backend reuse

### CODE REVIEW

File: packages/cdktf/lib/validations/validate-backends.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+): boolean {
+  // TODO: we could add `LocalBackend.isLocalBackend()` static helpers (breaking change!)
+  // and a isConflicting(otherBackend) method to our backends to move this logic into
+  // the individual backends instead of having it here
+
+  const ta = a.toTerraform(); // e.g. { terraform: { backend: { local: { path: 'terraform.tfstate' } } } }
+  const tb = b.toTerraform();
+
+  // do a deep compare for now
+  // TODO: in theory backends could be partly configured via environment variables which could make this a false positive
```
Comment (ansgarm) [inline packages/cdktf/lib/validations/validate-backends.ts:68]: I'm unsure whether we need to add a way to skip this validation (other than skipping all validations) for this case

Comment (DanielMSchmidt) [reply packages/cdktf/lib/validations/validate-backends.ts:68]: We could think about adding this in the user-code and in our templates to make skipping this easier and to educate folks about validations

**Review by DanielMSchmidt — DISMISSED** [review]
Let's make this a configuration option for the app so folks can disable this if they don't want it

**Review by mutahhir — APPROVED** [review]

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

**Pull request merged**